[[, [: add/update pages (#7530)
tldr-pages/tldr: pages/common/[[.md

# [[

> Check file types and compare values.
> Returns 0 if the condition evaluates to true, 1 if it evaluates to false.
> More information: <https://www.gnu.org/software/bash/manual/bash.html#Conditional-Constructs>.

- Test if two strings are equal/not equal:

> `[[ ${{variable}} {{==|=!}} "{{string}}" ]]`

Reviewer: ```suggestion
`[[ ${{variable}} {{==|!=}} "{{string}}" ]]`
```

- Test if a string conforms the specified glob/regex:

`[[ ${{variable}} {{==|=~}} {{pattern}} ]]`

- Test if two numbers are [eq]ual/[n]ot [e]qual/[g]reater [t]han/[l]ess [t]han/[g]reater than or [e]qual/[l]ess than or [e]qual:

`[[ ${{variable}} -{{eq|ne|gt|lt|ge|le}} "{{string}}" ]]`

> - Test if a the specified [f]ile exist:

Reviewer: ```suggestion
- Test if the specified [f]ile exists:
```

`[[ -f {{path/to/file}} ]]`

> - Test if a the specified [d]irectory exist:

Reviewer: ```suggestion
- Test if the specified [d]irectory exists:
```

`[[ -d {{path/to/directory}} ]]`

> - Test if a the specified file or directory [e]xist:

Reviewer: ```suggestion
- Test if the specified file or directory [e]xists:
```
